### PR TITLE
fix merge functionality for Descriptor objects

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

Addresses #144.

### CODE REVIEW

**Comment by dbecvarik** [comment]
I've already created different PR, with different aproach but similar outcome. 
https://github.com/jboss-container-images/concreate/pull/128

My approach is little different, it enables you fully override (not only concatenate list).

its based on assumption, that if we have list objects like:
``` yaml
list:
   - foo
   - bar
```
I enable alternative syntax like:
``` yaml
list:
   - name: foo # you dont need state:present here as this si default
   - name: bar
     state: absent
   - name: baz
     state: present
```
I support both syntax for all of our list object, so you can still use plain list form, but if you need an override (in overrides.yaml) you can use more complex syntax

**Comment by goldmann** [comment]
@rcernich @dbecvarik Can you guys talk to each other and decide on the approach?

File: concreate/tools.py [hunk 1]
```diff
@@ -100,7 +107,8 @@ def merge_lists(list1, list2):
             else:
                 for v1 in list1_dicts:
                     if v2['name'] == v1['name']:
-                        merge_dictionaries(v1, v2)
+                        if isinstance(v1, dict) and isinstance(v2, dict):
```
Comment (dbecvarik) [inline concreate/tools.py]: Name here is to be sure if we have list of dictionaries, that we are merging proper ones - thiw will just pair them in order, Im not sure if this works correctly.

**Comment by dbecvarik** [comment]
I've got inspired and implemented necessary parts to enable proper merging of descriptor based object in #147

**Comment by dbecvarik** [comment]
ITs probably not relevant anymore

**Pull request closed without merging**